NXP boards: nxp: frdm_ke16z: New board with 64kB Flash device added

mgalda82 opened the pull request:
boards: nxp: frdm_ke16z: New board with 64kB device added
Related to HAL_NXP: PR#595
https://github.com/zephyrproject-rtos/hal_nxp/pull/595

SOC_MKE16z4 and MKE15z4 (64kB, 48MHz) is now supported by DT
LPFLL clock option with proper frequency select
dts: nxp_ke1xz64.dtsi added.
lpfll_clk node defined in specific dts files, because of different frequency support
Tested on board: frdm_ke16z with "hello_world" and "blinky"

Signed-off-by: Michael Galda <michael.galda@nxp.com>

decsny reviewed (commented):
depending on how similar all the KE boards are, you might want to consider combining them all into the same folder and sharing files. Because the amount of files to maintain is getting out of hand otherwise if they are mostly the same board but with a very slightly different chip

dleach02 commented on soc/nxp/kinetis/ke1xz/soc.c:
```diff
@@ -81,31 +87,90 @@ static const scg_firc_config_t scg_firc_config = {
 	CLOCK_SetRunModeSysClkConfig(&scg_sys_clk_config);
 	do {
-		CLOCK_GetCurSysClkConfig(&current);
+	     CLOCK_GetCurSysClkConfig(&current);
```
The additional indent to CLOCK_GetCurSysClkConfig() needs to be another <tab> and not spaces.

decsny reviewed (dismissed):
waiting for answer about why new board folder is needed

dleach02 commented:
I'm fine with a board for this new FRDM board.

hakehuang committed:
dts: arm: nxp: nxp_ke1xz.dtsi: tsi0 & tsi1 nodes added

TSI (Touch Sensing IP) is now supported by DT
for the Kinetis KE1x family
dts: nxp_ke1xz, nxp_ke17z and nxp_ke17z512.dtsi
Tested on freedom boards

Signed-off-by: Michael Galda <michael.galda@nxp.com>
Signed-off-by: Hake Huang <hake.huang@nxp.com>

hakehuang committed:
drivers: clock_control: kinetis scg support

add driver and dts-binding for kinetis_scg driver
used on ke1xz

Signed-off-by: Hake Huang <hake.huang@nxp.com>
Signed-off-by: Michael Galda <michael.galda@nxp.com>

hakehuang committed:
soc: nxp: kinetis: ke1xz: add LPFLL clock support to SCG

LPFLL (soc: scg: "lpfll_clk") clock source can be now selected
in KE1xz Device tree (nxp_ke1xz.dtsi., board.dts or overlay)
Tested on boards: frdm_ke15z, frdm_ke17z, frdm_ke17z512

Signed-off-by: Michael Galda <michael.galda@nxp.com>
Signed-off-by: Hake Huang <hake.huang@nxp.com>

hakehuang committed:
boards: nxp: frdm_ke16z: New board with 64kB device added

SOC_MKE16z4 and MKE15z4 (64kB, 48MHz) is now supported by DT
LPFLL clock option with proper frequency select
dts: nxp_ke1xz64.dtsi added.
Tested on board: frdm_ke16z

Signed-off-by: Michael Galda <michael.galda@nxp.com>
Signed-off-by: Hake Huang <hake.huang@nxp.com>
